File: I1_MVPAndRetrofit/app/src/main/java/com/example/tsvetelinastoyanova/weatherreportrevisited/weatherdetails/WeatherDetailsContract.java
package com.example.tsvetelinastoyanova.weatherreportrevisited.weatherdetails;

import com.example.tsvetelinastoyanova.weatherreportrevisited.BasePresenter;
import com.example.tsvetelinastoyanova.weatherreportrevisited.BaseView;
import com.example.tsvetelinastoyanova.weatherreportrevisited.model.WeatherObject;

public interface WeatherDetailsContract {

    interface View extends BaseView<Presenter> {
        void changeView(WeatherObject weatherObject);
    }

    interface Presenter extends BasePresenter {

    }
}

File: I1_MVPAndRetrofit_Kotlin/app/src/main/java/com/example/tsvetelinastoyanova/weatherapp/weatherdetails/WeatherDetailsContract.kt
package com.example.tsvetelinastoyanova.weatherapp.weatherdetails

import com.example.tsvetelinastoyanova.weatherapp.BasePresenter
import com.example.tsvetelinastoyanova.weatherapp.BaseView
import com.example.tsvetelinastoyanova.weatherapp.model.currentweather.CurrentWeatherObject

interface WeatherDetailsContract {

    interface View : BaseView<Presenter> {
        fun changeView(currentWeatherObject: CurrentWeatherObject)
    }

    interface Presenter : BasePresenter
}